### DISCUSSION TITLE
Is it possible for data from `useQuery` to be undefined when isSuccess === true?

### DISCUSSION THREAD

**Discussion by lisherwin** [TanStack/query#8187] [Q&A] [answered]
Hi! I've been using React Query for two years and I'm stumped.

I recently encountered a crash log where my query’s `isSuccess` is `true`, but the `data` returned from `useQuery` is `undefined`. Is there a edge case that I'm missing where this could happen? How I understand it is that if `isSuccess` is true, then that means the data is loaded and not errored out, and that these are mutually exclusive states. I am checking for data nullity in a dependent query.

Sorry I'm not able to reproduce this issue (if I was able to, then no need for this question :))

Our code looks something like this:
```
export function useDataB() {
  const { isSuccess: isSuccessDataA, data: dataA } = useDataA();

  const storage = useStorage();

  const queryKey = "data-b";
  const queryFn = async (): Promise<B> => {
    if (!dataA) throw new Error("Missing A"); // this throws for some reason
    const dataB = await storage.get(...);
     ...
  };

  return useQuery({
    enabled: isSuccessDataA,
    queryKey,
    queryFn,
  });
}
```

Thank you in advance!

**Comment by TkDodo** [comment, marked as answer]
Only if you return undefined from your queryFn, which isn't allowed.
